From 3a1d6cd2f41ecd73c7d867b22546857286db8f95 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Wed, 26 Apr 2023 17:01:31 +0800
Subject: [PATCH] Do not create user and group

Do not create user/group since we are in cross build env.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 Makefile.am  |  2 +-
 configure.ac | 11 -----------
 2 files changed, 1 insertion(+), 12 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index f03d15c..30fad17 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -994,7 +994,7 @@ endif
 	@test -n "$(DESTDIR)" || $(call check_fapi_dirs)
 endif
 
-install-data-hook: install-dirs
+install-data-hook:
 	-if [ ! -z "$(udevrulesprefix)" ]; then \
 		mv $(DESTDIR)$(udevrulesdir)/tpm-udev.rules $(DESTDIR)$(udevrulesdir)/$(udevrulesprefix)tpm-udev.rules; \
 	fi
diff --git a/configure.ac b/configure.ac
index eb6051e..a52c472 100644
--- a/configure.ac
+++ b/configure.ac
@@ -663,17 +663,6 @@ AM_CONDITIONAL([SYSD_SYSUSERS], [test "x$systemd_sysusers" = "xyes" && test "x$s
 AC_CHECK_PROG(systemd_tmpfiles, systemd-tmpfiles, yes)
 AM_CONDITIONAL([SYSD_TMPFILES], [test "x$systemd_tmpfiles" = "xyes" && test "x$tmpfilesdir" != "xno"])
 
-# Check all tools used by make install
-AS_IF([test "$HOSTOS" = "Linux" && test "x$systemd_sysusers" != "xyes"],
-    [ AC_CHECK_PROG(useradd, useradd, yes)
-      AC_CHECK_PROG(groupadd, groupadd, yes)
-      AC_CHECK_PROG(adduser, adduser, yes)
-      AC_CHECK_PROG(addgroup, addgroup, yes)
-      AS_IF([test "x$addgroup" != "xyes" && test "x$groupadd" != "xyes" ],
-         [AC_MSG_ERROR([addgroup or groupadd are needed.])])
-      AS_IF([test "x$adduser" != "xyes" && test "x$useradd" != "xyes" ],
-         [AC_MSG_ERROR([adduser or useradd are needed.])])])
-
 AC_SUBST([PATH])
 
 dnl --------- Doxy Gen -----------------------
-- 
2.25.1

